src/routes.js

```diff
@@ -44,4 +45,25 @@ router.post('/coffeegrams', async ctx => {
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
+router.post('/users/:username', async ctx => {
+  var user = await User.findOne({ username: ctx.params.username });
+  if (!user) {
+    throw new NotFound();
+  }
+  if (user.id !== ctx.state.currentUser.id) {
+    ctx.request.flash('error', "You\'re not allowed to edit someone else\'s profile");
+    return ctx.redirect('back');
+  }
+
+  var { bio } = ctx.request.body;
+  clean(bio);
+  if (isBlank(bio)) {
+    ctx.request.flash('error', "Your bio looks empty!");
+    return ctx.redirect('back');
+  }
+
+  await User.findByIdAndUpdate({ _id: user.id }, { $set: { bio }});
```

Comment:
Now I think you are ready to move away from ^^. Since you have loaded the `user` to a local variable, now you just do:

```javascript
user.bio = bio;
await user.save();
```